src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -186,113 +185,96 @@ private bool RemoveSession(int sessionId, out ProfilerSession session)
-        {
-            var events = new List<ProfilerEvent>();
-            try
+            foreach (var kvp in xEvent.Actions)
             {
-                if (session == null || session.XEventSession == null)
-                {
-                    return events;
-                }
-
-                var targetXml = session.XEventSession.GetTargetXml();
+                profileEvent.Values.Add(kvp.Key, kvp.Value.ToString());
+            }
+            var eventList = new List<ProfilerEvent>();
+            eventList.Add(profileEvent);
+            var eventsLost = session.EventsLost;
 
-                XmlDocument xmlDoc = new XmlDocument();
-                xmlDoc.LoadXml(targetXml);
+            if (eventList.Count > 0 || eventsLost)
+            {
+                session.FilterOldEvents(eventList);
+                eventList = session.FilterProfilerEvents(eventList);
+                // notify all viewers of the event.
+                List<string> viewerIds = this.sessionViewers[session.XEventSession.Id];
 
-                var nodes = xmlDoc.DocumentElement.GetElementsByTagName("event");
-                foreach (XmlNode node in nodes)
+                foreach (string viewerId in viewerIds)
                 {
-                    var profilerEvent = ParseProfilerEvent(node);
-                    if (profilerEvent != null)
+                    if (allViewers[viewerId].active)
                     {
-                        events.Add(profilerEvent);
+                        SendEventsToListeners(viewerId, eventList, eventsLost);
                     }
                 }
             }
-            catch (XEventException)
-            {
-                SendStoppedSessionInfoToListeners(session.XEventSession.Id);
-                ProfilerSession tempSession;
-                RemoveSession(session.XEventSession.Id, out tempSession);
-            }
-            catch (Exception ex)
-            {
-                Logger.Write(TraceEventType.Warning, "Failed to poll session. error: " + ex.Message);
-            }
-            finally
+        }
+
+         /// <summary>
+        /// Nulls out properties in ConnectionDetails that aren't compatible with XElite.
+        /// </summary>
+        private ConnectionDetails CreateXEliteConnectionDetails(ConnectionDetails connDetails){
+            connDetails.ConnectRetryCount = null;
+            connDetails.ConnectRetryInterval = null;
+            connDetails.MultiSubnetFailover = null;
+            return connDetails;
+        }
+
+        /// <summary>
+        /// Creates a stream for new XEvents if the session hasn't streamed yet.
+        /// </summary>
+        private void ProcessStream(int id, ProfilerSession session)
+        {
+            CancellationTokenSource threadCancellationToken = new CancellationTokenSource();
+            ConnectionDetails trimmedDetails = CreateXEliteConnectionDetails((session.XEventSession as XEventSession).ConnInfo.ConnectionDetails);
```

Comment:
Why does the XEventSession not have a connection details that already can be used for xelite connections? It seems like we should be doing this as soon as the session is created.

Reply:
The ConnectionInfo class currently does not allow for properties such as connection details to be altered after they are created, the set function for it is private. Unless we want to change the class to allow for that, you have to copy the ConnectionDetails and clean it separately.

Reply:
Yes, you'd create a new ConnectionDetails at some point. I'm saying that when the `ProfilerSession` is created then the connection info it creates there calls `CreateXEliteConnectionDetails` so that this object already has the expected details and no further processing is needed.

Reply:
The ConnectionInfo class for profiler is retrieved by calling ConnectionServiceInstance.tryfindconnection which uses a trygetvalue on a map, if we want to alter the creation of the connectioninfo's connectiondetails as its being created, it would require alterations to the connection service, so it's not practical there, if we want to store the connection details in XEventSessions, that might be a better option.